app/src/release/java/com/futurice/freesound/app/module/InstrumentationModule.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.app.module;
+
+import com.futurice.freesound.network.api.FreeSoundApiInterceptor;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import okhttp3.Interceptor;
+import okhttp3.logging.HttpLoggingInterceptor;
+
+import static com.futurice.freesound.app.module.ApiModule.API_TOKEN_CONFIG;
+
+@Module
+public final class InstrumentationModule {
+
+    @Provides
+    @Singleton
+    @ApiModule.NetworkInterceptors
+    static List<Interceptor> provideNetworkInterceptors(FreeSoundApiInterceptor apiInterceptor,
+                                                        HttpLoggingInterceptor loggingInterceptor) {
+        List<Interceptor> networkInterceptors = new ArrayList<>(1);
+        networkInterceptors.add(apiInterceptor);
+        return networkInterceptors;
+    }
+
+    @Provides
+    @Singleton
+    @ApiModule.AppInterceptors
+    static List<Interceptor> provideAppInterceptors() {
+        return Collections.emptyList();
+    }
+
+    @Provides
+    @Singleton
+    static FreeSoundApiInterceptor provideApiInterceptor(@Named(API_TOKEN_CONFIG) String apiToken) {
+        return new FreeSoundApiInterceptor(apiToken);
+    }
+
+    @Provides
+    @Singleton
+    static HttpLoggingInterceptor provideLoggingInterceptor() {
+        return null;
```

Comment:
I think it's better not to return `null` from Dagger module provide methods because exposing `null` gets messy when injected and there are ways to structure your modules to support optional dependencies per build type.

In this case, I think the `InstrumentationModule` should expose a provider of `List<Interceptor>` which the `ApiModule` uses to define its `OkHttpClient`. That way you can expose an empty list from `release` and a non-empty list from `debug`.